<?xml version="1.0"?>
<launch>

  <include file="$(find rslidar_sdk)/launch/start.launch" />

  <arg name="img_topic" default="lbas_image" />
  <arg name="max_freq" default="90.0" />
  <node pkg="camera_lbas" name="camera_lbas_node" type="camera_lbas_node" output="screen">
    <param name="if_auto" value="true" type="bool" />
    <param name="if_resize" value="false" type="bool" />
    <param name="exp_time" value="5000.0" type="double" />
    <param name="ip" value="192.168.1.201" type="string" />
    <param name="max_hz" value="$(arg max_freq)" type="double" />
    <!-- do not modify the above param except the IP of camera is really changed -->
    <param name="pubtopic_name" value="$(arg img_topic)" type="string" />
    <param name="display" value="0" type="int" />
  </node>

  <!-- Load Parameter -->
  <param name="yolov5_path" value="$(find yolov5_ros)/yolov5" />
  <param name="use_cpu" value="false" />
  <!-- Start yolov5 and ros wrapper -->
  <node pkg="yolov5_ros" type="yolo_v5.py" name="yolov5_ros" output="screen">
    <param name="weight_path" value="$(find yolov5_ros)/weights/yolov5s.pt" />
    <param name="image_topic" value="/lbas_image" />
    <param name="pub_topic" value="/yolov5/BoundingBoxes" />
    <param name="camera_frame" value="camera_color_frame" />
    <param name="conf" value="0.3" />
  </node>

  <node pkg="lidar_camera_projection" type="projector" name="projector" output="screen">
    <rosparam command="load" file="$(find lidar_camera_projection)/config/params.yaml" />
  </node>
</launch>